goaci.go

```diff
@@ -38,45 +41,75 @@ func debug(i ...interface{}) {
 	}
 }
 
+type StringVector []string
+
+func (v *StringVector) String() string {
+	return "\"" + strings.Join(*v, "\" \"") + "\""
+}
+
+func (v *StringVector) Set(str string) error {
+	*v = append(*v, str)
+	return nil
+}
+
 func main() {
+	var (
+		execOpts            StringVector
+		goDefaultBinaryDesc string
+		goBinaryOpt         string
+		goPathOpt           string
+	)
+
+	// Find the go binary
+	gocmd, err := exec.LookPath("go")
+
+	flag.Var(&execOpts, "exec", "Parameters passed to app, can be used multiple times")
+	if err != nil {
+		goDefaultBinaryDesc = "Go binary to use (default: none found in $PATH, so it must be provided)"
+	} else {
+		goDefaultBinaryDesc = "Go binary to use (default: whatever go in $PATH)"
+	}
+	flag.StringVar(&goBinaryOpt, "go-binary", gocmd, goDefaultBinaryDesc)
+	flag.StringVar(&goPathOpt, "go-path", "", "Custom GOPATH (default: a temporary directory)")
+	flag.Parse()
 	if os.Getenv("GOPATH") != "" {
-		die("to avoid confusion GOPATH must not be set")
+		warn("GOPATH envvar is ignored, use --go-path=\"$GOPATH\" option instead")
```

Comment:
> And about GOROOT - is there any point in using go 1.3 and specifying GOROOT pointing to root of go 1.4?

Not sure exactly what you mean here, they should definitely not be different versions

> I think that better idea is to point goaci to using specific go binary, which already knows that it should use its own GOROOT (via go env). Also, requiring specifying GOROOT is bothersome and shouldn't be needed. I don't have it set in my environment.

For anyone using a precompiled version of go that's not installed in the default directory (which is not necessarily uncommon), `GOROOT` is still required. I use it, for one ;-). So `--go-binary` alone is not sufficient:

```
: trevize goaci endocode-krnowak/work ; bin/goaci  --go-binary=/home/jon/go/bin/go github.com/coreos/etcd
GOROOT envvar is ignored, use --go-binary="$GOROOT/bin/go" option instead
go: cannot find GOROOT directory: /usr/local/go
error running go: exit status 2
```